front/src/modules/simulationResult/components/SpeedSpaceChart/utils.ts

```diff
@@ -56,3 +69,164 @@ export const createSlopeCurve = (slopes: GradientPosition[], data: number[]): He
     height: (step.height * referentialHeight) / dataHeight,
   }));
 };
+
+/**
+ * Create the altitude curve based from the slopes data
+ */
+export const createSlopeCurveV2 = (
+  slopes: PositionData<'gradient'>[],
+  baseSpeeds: number[]
+): HeightPosition[] => {
+  const slopesCurve: HeightPosition[] = [];
+  slopes.forEach((step, idx) => {
+    if (idx % 2 === 0 && slopes[idx + 1]) {
+      if (idx === 0) {
+        slopesCurve.push({ height: 0, position: step.position });
+      } else {
+        const distance = step.position - slopesCurve[slopesCurve.length - 1].position;
+        const height =
+          (distance * slopes[idx - 2].gradient) / 1000 + slopesCurve[slopesCurve.length - 1].height;
+        slopesCurve.push({ height, position: step.position });
+      }
+    }
+  });
+  const referentialHeight = calculateReferentialHeight(baseSpeeds);
+  const maxRadius = d3.max(slopesCurve.map((step) => step.height));
+  const minRadius = d3.min(slopesCurve.map((step) => step.height));
+  let dataHeight = 0;
+  if (maxRadius !== undefined && minRadius !== undefined) {
+    dataHeight = maxRadius - minRadius;
+  }
+  return slopesCurve.map((step) => ({
+    ...step,
+    height: (step.height * referentialHeight) / dataHeight,
+  }));
+};
+
+export const createCurvesCurveV2 = (
```

Comment:
Same as createCurveCurve
```suggestion
```

Reply:
Same as interpolateOnPositionV2, PositionData<'gradient'>[] and RadiusPosition[] types are the same but I thought it would be better to create another function for the moment when we might drop all RadiusPosition types etc ?